Sources/SwiftGRPCNIO/GRPCClientResponseChannelHandler.swift

```diff
@@ -14,34 +14,27 @@
  * limitations under the License.
  */
 import Foundation
+import SwiftProtobuf
 import NIO
 import NIOHTTP1
-import SwiftProtobuf
 
-/// The final client-side channel handler.
+/// A base channel handler for receiving responses.
 ///
-/// This handler holds promises for the initial metadata and the status, as well as an observer
-/// for responses. For unary and client-streaming calls the observer will succeed a response
-/// promise. For server-streaming and bidirectional-streaming the observer will call the supplied
-/// callback with each response received.
-///
-/// Errors are also handled by the channel handler. Promises for the initial metadata and
-/// response (if applicable) are failed with first error received. The status promise is __succeeded__
-/// with the error as the result of `GRPCStatusTransformable.asGRPCStatus()`, if available.
-/// The stream is also closed and any inbound or outbound messages are ignored.
-internal class GRPCClientChannelHandler<RequestMessage: Message, ResponseMessage: Message> {
+/// This includes hold promises for the initial metadata and status of the gRPC call. This handler
+/// is also responsible for error handling, via an error delegate and by appropriately failing the
+/// aforementioned promises.
+internal class GRPCClientResponseChannelHandler<ResponseMessage: Message>: ChannelInboundHandler {
+  public typealias InboundIn = GRPCClientResponsePart<ResponseMessage>
+  public typealias OutboundIn = Any
```

Comment:
Could you explain this?

Reply:
It wasn't meant to be there, thanks for catching!